Add a markmap mind map page for scaling InnerSource

jeffabailey opened the pull request:
An output of an @InnerSourceCommons/ispo-working-group meeting: https://docs.google.com/document/d/12PFkoiaukmrMbnTPYKERzViKCULfClutXN4ydVPv3H0/edit

feat: add markmap partial
feat: add markmap shortcode
feat: add markmap friendly markmap content
fix: skip yourmax processing if no items defined

jeffabailey committed:
feat: add markmap partial
feat: add markmap shortcode
feat: add markmap friendly markmap content
fix: skip yourmax processing if no items defined

jeffabailey committed:
chore: add a solid border around the map

dellagustin-sap commented:
Hi @jeffabailey , I tested the PR locally and found two issues:
1. Major - the /learn page is showing some "weird" stuff, screenshot below, it shows the markdown text of the mindmap under "Scaling InnerSource" and it shows also an additional section with the content of `markmap_content.md`, but slightly different.
2. Minor - On the scaling page, the mindmap has a kind of a viewport, where it can be zoomed in and out, panned and scrolled, but without scrollbars or boarders - this is more a weird feature than a bug, just wanted to mention

I found a way to fix 1 and I'll send a PR on top of this PR with a fix.

<img width="296" alt="image" src="https://github.com/user-attachments/assets/db17c6e3-530b-448e-a9c6-2577b9d5b8e6">

<img width="609" alt="image" src="https://github.com/user-attachments/assets/ad0b789a-1d56-493a-a715-bae23bed391a">

Commit:
fix: Scaling content on Learn page

This commit fixes an issue on the `/learn` page introduced with
the mindmap for scaling.

Acording to my tests:
1. If a folder does not have a `_index.md` file, every file under
   the folder will generate a section under the `/learn` page. in
   This way, both files, `mindmap.md` and `markmap_content.md`
   were generating content. To fix this, I renamed `mindmap.md`
   to `_index.md`.
2. When a file does not mave a `description` property in the
   frontmatter block, it will use the content of the file as
   description. To fix this I added a `description` property to
   `mindmap.md` (now renamed to `_index.md`.

Related issues
- https://github.com/InnerSourceCommons/innersourcecommons.org/pull/902#issuecomment-2422746281

dellagustin-sap commented:
@jeffabailey , here is the proposed fix: https://github.com/InnerSourceCommons/innersourcecommons.org/pull/902

dellagustin-sap commented on content/en/learn/scaling/mindmap.md:
```diff
@@ -0,0 +1,9 @@
+weight: 1
+---
+
+{{< markmap path="markmap_content.md" >}}
```
@jeffabailey all the other files on the repo use a `-` as separator.
I suggest renaming `markmap_content.md` to `markmap-content.md` for consistency.

dellagustin-sap commented:
Also, it would be nice if we could make the links open in a new tab, but I think that would require some implementation.
Here are some refs:
- https://github.com/markmap/markmap/blob/master/packages/markmap-autoloader/README.md
- https://github.com/markmap/markmap/issues/129

by adding a transformer plugin with something like this, we may be able to do it:
```
<script>
  window.markmap = {
    autoLoader: {
      transformPlugins: [...markmap.builtInPlugins, { 
     name: 'target-blank', 
     transform(transformHooks) { 
       transformHooks.parser.tap((md) => { 
         md.renderer.renderAttrs = wrapFunction( 
           md.renderer.renderAttrs, 
           (renderAttrs, token) => { 
             let attrs = renderAttrs(token); 
             if (token.type === 'link_open') { 
               attrs += ' target="_blank"'; 
             } 
             return attrs; 
           }, 
         ); 
       }); 
       return {}; 
     }, 
   }],
    },
  };
</script>
```

I'll need to test this later.

jeffabailey committed:
Merge pull request #902 from dellagustin-sap/pr#901-fix-learn-page-new-scaling-content

fix: Scaling content on Learn page

jeffabailey committed:
chore: use hyphen for consistency

jeffabailey committed:
fix: include the renamed md file

dellagustin-sap commented:
LGTM, but I'm not a maintainer.
A minor quirk, viewing on my 15" macbook display, the viewport of the mindmap does not fit the screen in 100% zoom, vertically.
<img width="1710" alt="image" src="https://github.com/user-attachments/assets/16b4063e-24c2-4995-bde2-13a57e3ce6ba">

jeffabailey commented:
Unless we compact the mindmap, I'm afraid that's an avoidable quirk.

However, assuming that all operating systems and browsers allow you to hold ctrl to pan the mindmap in any direction, we could put a quick instruction above the mindmap.

dellagustin-sap commented:
> However, assuming that all operating systems and browsers allow you to hold ctrl to pan the mindmap in any direction, we could put a quick instruction above the mindmap.

I was thinking about that too, some instructions on top would be nice. I don't think we need to hold the merge for that though, that could be an increment with another PR, unless you want to include it already, up to you @jeffabailey .

One minor ask, can you please link to https://docs.google.com/document/d/12PFkoiaukmrMbnTPYKERzViKCULfClutXN4ydVPv3H0/edit on the description of the PR? Just for additional traceability on the origin of the mindmap.

jeffabailey commented:
Hello @InnerSourceCommons/website-maintainers do you have any concerns with merging this PR?

If no, please approve. :)

jeffabailey committed:
feat: add content from WG session
feat: start map from level 2
feat: add complete Miro content
